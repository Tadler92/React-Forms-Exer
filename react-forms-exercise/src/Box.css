.Box {
  display: flex;
  align-items: center;
}

.Box div {
  margin: 30px 15px;
}

.Box button {
  background-color: crimson;
  color: white;
  padding: 10px 15px;
  border-radius: 40px;
}