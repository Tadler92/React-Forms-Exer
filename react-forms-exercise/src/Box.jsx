import './Box.css'
const Box = ({backgroundColor, width, height, removeBox, id}) => {
  const remove = () => removeBox(id);
  return (
    <div className="Box">
      <div style={{
        backgroundColor: `${backgroundColor}`,
        width: `${width}px`,
        height: `${height}px`
      }}></div>
      <button onClick={remove}>X</button>
    </div>
  )
}


export default Box;