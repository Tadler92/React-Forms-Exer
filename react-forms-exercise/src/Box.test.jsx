import {render, fireEvent, waitFor} from "@testing-library/react";
import Box from "./Box";
import { expect, it } from "vitest";


describe("Box Component", () => {
  it("renders without crashing", function () {
    render(<Box />);
  });
  
  
  it("matches snapshot", function () {
    const {asFragment} = render(<Box />);
    expect(asFragment()).toMatchSnapshot();
  });
});