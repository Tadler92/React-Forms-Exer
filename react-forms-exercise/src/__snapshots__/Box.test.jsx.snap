// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Box Component > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Box"
  >
    <div />
    <button>
      X
    </button>
  </div>
</DocumentFragment>
`;
